lib/http/node.js

```diff
@@ -76,15 +75,40 @@ AWS.NodeHttpClient = AWS.util.inherit({
 
   writeBody: function writeBody(stream, httpRequest) {
     var body = httpRequest.body;
-
-    if (body && WritableStream && ReadableStream) { // progress support
-      if (!(body instanceof Stream)) body = AWS.util.buffer.toStream(body);
-      body.pipe(this.progressStream(stream, httpRequest));
-    }
+    var totalBytes = parseInt(httpRequest.headers['Content-Length'], 10);
+    var loadedBytes = 0;
 
     if (body instanceof Stream) {
+      // for progress support of streaming content
+      // tap the data event of the stream in addition to piping
+      body.on('data', function(chunk) {
```

Comment:
I think tapping into the `data` event on a stream could cause some unintended side-effects. In node.js 0.10.x, binding a `data` listener on a stream will cause the stream to emit 'data' events as fast as it can, ignoring the back-pressure that's automatically in place when using `pipe`. If the writable stream is slow, then this could cause loss of data.

I don't think this is an issue in versions of node.js >= 0.12.x, and some simple testing seems to confirm that. However, we need to work with node.js 0.10.x as well.

The current method creates a new `writable` stream that also gets piped into in order to emit the 'sendProgress' events. I know it'll require refactoring your logic but that path seems safer across all our supported versions of node.